apps/omg_child_chain/test/omg_child_chain/block_queue/core_test.exs

```diff
@@ -89,173 +67,124 @@ defmodule OMG.ChildChain.BlockQueue.CoreTest do
       assert [1000, 2000, 3000] == Core.child_block_nums_to_init_with(0, 3000, @child_block_interval, 2)
       assert [2000, 3000, 4000, 5000] == Core.child_block_nums_to_init_with(4000, 5000, @child_block_interval, 2)
     end
+  end
 
+  describe "new/1" do
     test "Recovers after restart to proper mined height" do
       assert [%{hash: "8", nonce: 8}, %{hash: "9", nonce: 9}] =
-               [{5000, "5"}, {6000, "6"}, {7000, "7"}, {8000, "8"}, {9000, "9"}]
-               |> recover(7000)
+               Core.new(
+                 7000,
+                 [{5000, "5"}, {6000, "6"}, {7000, "7"}, {8000, "8"}, {9000, "9"}],
+                 "7",
+                 10,
+                 child_block_interval: @child_block_interval
+               )
                |> elem(1)
                |> Core.get_blocks_to_submit()
     end
 
     test "Recovers after restart and talking to an un-synced geth" do
       # imagine restart after geth is nuked and hasn't caught up
-      # testing against a disaster scenario where `BlockQueue` would start pushing old blocks again
-      finality_threshold = 12
-      mined_blknum = 6000
-      range = Core.child_block_nums_to_init_with(mined_blknum, 9000, @child_block_interval, finality_threshold)
-      known_hashes = ~w(1 2 3 4 5 6 7 8 9)
-
+      # testing protecting against a disaster scenario, where `BlockQueue` would start pushing old blocks again
       {:ok, state} =
         Core.new(
-          mined_child_block_num: mined_blknum,
-          known_hashes: Enum.zip(range, known_hashes),
-          top_mined_hash: "6",
-          parent_height: 6,
-          child_block_interval: @child_block_interval,
-          chain_start_parent_height: 1,
-          block_submit_every_nth: 1,
-          finality_threshold: finality_threshold,
-          last_enqueued_block_at_height: 0
+          6000,
+          [{5000, "5"}, {6000, "6"}, {7000, "7"}, {8000, "8"}, {9000, "9"}],
+          "6",
+          10,
```

Comment:
Is there a reason we moved away from the named arguments? I think it helped make it clear what the arguments represented and how to invoke `Core.new` here.

Reply:
ah, so I liked those named args here too, but:
 - there were overall too many of those, and I wanted a clear and standard distinction between required and optional
 - I tried sth like `def new([required_1: ..., required_2: ... | optional])` to have best of both, but it seems not not be supported and the alternative: `def new([{:required_1, _}, {:required_2, _} | optional` either didn't work or it seemed weird and nonstandard. 
 - And then just `def new(opts \\ [])` and have the required args enforced in the body didn't click too well neither. But maybe this could be an alternative, it just would hide the required vs optional distinction.

I think 4 unnamed required params and then options is pretty standard. I can try the last `def new(opts \\ [])` if you like and see how it goes.

Reply:
ok, did that, I think it looks better ff2a127946a4c8de40a752a1d2827674056c4788, thx!